ingredient/ingredient-postgresql/src/functions.ts

```diff
@@ -1,11 +1,72 @@
 import {BaseUtility, IngredientManager} from '@azbake/core'
+import { NetworkManagementClient } from '@azure/arm-network';
+import { Subnet, VirtualNetwork } from '@azure/arm-network/esm/models';
+import { PrivateDnsManagementClient, PrivateZone } from '@azure/arm-privatedns';
+import { DefaultAzureCredential, ClientSecretCredential,ChainedTokenCredential } from "@azure/identity";
+import { RestError } from '@azure/ms-rest-js';
 
 export class PostgreSQLDBUtils extends BaseUtility {
-
+    private token = new ClientSecretCredential(this.context.AuthToken.domain, this.context.AuthToken.clientId, this.context.AuthToken.secret);
+    private credential = new ChainedTokenCredential(this.token, new DefaultAzureCredential());
+    
     public create_resource_name(): string {
-        let util = IngredientManager.getIngredientFunction("coreutils", this.context);
+        const util = IngredientManager.getIngredientFunction("coreutils", this.context);
         const name = util.create_resource_name("pgsql", null, true);
         return name;
     }
-}
 
+    public create_resource_uri(access: string): string {
+        const infix = (access == 'private') ? '.private' : '';
+        return `${this.create_resource_name()}${infix}.postgres.database.azure.com`; 
+    }
+
+    public async get_resource_group(): Promise<string> {
```

Comment:
what's the use of this? Why replicate/passthrough to a coretutils function. Just call coreutils to get the RG wherever you're calling this.. coreutils.resource_group() has more options as well.

Reply:
Oops, this actually doesn't even get called. I must have copied it from the apim ingredient to use as a template and then forgotten to remove it. I'll remove it now.